### DISCUSSION TITLE
Filtering supabase timestamps in Nextjs

### DISCUSSION THREAD

**Discussion by JustinBrenner** [supabase/supabase#8760] [Questions] [answered]
Hi All, 

I am trying to filter some records based on a range that a timestamp was generated. 

I have referenced the documentation and tried using .gt() & .lt() along with some other things that i found here on this other discussion: https://github.com/supabase/supabase/discussions/3734 but i am still unable to get the result that i am looking for. 

Basically i am just trying to filter records created on the previous month x/xx/xxxx 00:00:00 and i am just stumped on it. 

Here is the code that i am using right now: 

    const date = new Date();
    const firstDayPrevMonth = new Date(date.getFullYear(), date.getMonth() - 1, 1);
    console.log(firstDayPrevMonth);
    
    const lastDayPrevMonth = new Date(date.getFullYear(), date.getMonth(), 0);
    console.log(lastDayPrevMonth);
    
    const {data, error } = await supabase.from('userSearch').select('*').filter('created_at', 'gt', firstDayPrevMonth);

**Comment by GaryAustin1** [comment, marked as answer]
Looks like you need to use firstDayPrevMonth.toISOString() and then chain your less than filter also.
Also it might be cleaner to just use .gt('col', time).lt('col',time).  I don't think filter is needed for any formatting reasons.

**Reply by JustinBrenner** [reply]
Thanks so much this helped a ton, now one more question for yah. When i use the above code as you have it written i only return a subset of the timestamp data but not all of the data is returned it only is returning from the first to around the 11th of the month. Kind of strange? or am i still missing something?

**Reply by GaryAustin1** [reply]
It would help to show your "working" code.  Also the default limit of rows to return is 1000, if you are returning more than that you have to paginate or set a larger count in API settings.

**Reply by JustinBrenner** [reply]
You are right, i apparently have been looking at my code too much today without taking necessary breaks to reset a little :D After i asked i remembered the API limitations. 

Again, Gary really appreciate the help, you are awesome!
